wolfcrypt/src/aes.c

```diff
@@ -3601,6 +3601,7 @@ int wc_AesGcmSetKey(Aes* aes, const byte* key, word32 len)
     if (!((len == 16) || (len == 24) || (len == 32)))
         return BAD_FUNC_ARG;
 
+    XMEMSET(aes, 0, sizeof(Aes));
```

Comment:
@tmael : This will not work. Instead the `wc_AesInit` should be called before `wc_AesGcmSetKey`. The reason is `wc_AesInit` can optionally populate `heap` and `devId`, which your code would override.

Reply:
@dgarske Good point! Fixed. 
I'm keeping the init in wc_AesGcmSetKey() to handle cases where wc_AesInit() wasn't called before. For example,
wolfssl-examples/crypto/aes/aes-file-encrypt.c missed calling wc_AesInit() before wc_AesGcmSetKey().